src/main/java/net/floodlightcontroller/statistics/StatisticsCollector.java

```diff
@@ -162,6 +171,107 @@ protected long getSpeed(NodePortTuple npt) {
 		}
 
 	}
+	
+	
+	/*
+	 * 	Added by Charlie Hardwick-Kelly Carleton University 
+	 * 	The purpose of this class is to track per flow metrics from within the network  
+	 * 	one key advantage of the flow metrics is that they contain a duration time stamp 
+	 * 	this eliminates the error involved from the control plane and allows us to compute the bandwidth of 
+	 * 	a particular flow with greater accuracy
+	 * 
+	 *  	Steps 
+	 * 		1. issue a meter stats request to all switches in parallel 
+	 * 		2. Analyze the variables 
+	 * 		3. if valid calculate relevant bandwidth 
+	 * 		4. store information in hash 
+	 * 		5. send information to the gui.
+	 * 
+	 */
+	protected class FlowStatsCollector implements Runnable {
+
+		@Override
+		public void run() {
+			Map<DatapathId, List<OFStatsReply>> flowReplies = getSwitchStatistics(switchService.getAllSwitchDpids(), OFStatsType.FLOW);
+			if(debug) {
+				System.out.print("\033[H\033[2J");// add this to clear the command line every round only works on real terminal
```

Comment:
I would probably get rid of this, especially if we plan on adding to GUI anyways